ui-tests/tst/software/aws/toolkits/jetbrains/uitests/tests/SamRunConfigTest.kt

```diff
@@ -86,17 +81,55 @@ class SamRunConfigTest {
+                        // joining all the text it has into a string
+                        val fixture = findRunDialog().find<ContainerFixture>(byXpath("//div[@class='MyEditorTextField']"))
+                        assertThat(fixture.findAllText().joinToString("") { it.text }).isEqualTo(input)
+                    }
+                    pressOk()
+                }
+                step("Setup handler based run configuration") {
+                    step("Reopen the run configuration menu") {
+                        findAndClick("//div[@accessiblename='[Local] SomeFunction']")
+                        find<JListFixture>(byXpath("//div[@class='MyList']")).selectItem("Edit Configurations...")
+                    }
+                    addRunConfig()
+                    find<ComboBoxFixture>(byXpath("//div[@text='Runtime:']/following-sibling::div[@class='ComboBox']")).selectItem("java11")
+                    findAndClick("//div[@class='HandlerPanel']")
+                    keyboard { enterText("helloworld.App::handleRequest") }
+                    findAndClick("//div[@class='MyEditorTextField']")
+                    keyboard { enterText(input) }
+                    pressOk()
+                }
+                step("Validate handler run configuration was saved and loads properly") {
+                    step("Reopen the run configuration") {
+                        findAndClick("//div[@accessiblename='[Local] App.handleRequest']")
+                        find<JListFixture>(byXpath("//div[@class='MyList']")).selectItem("Edit Configurations...")
+                    }
+                    step("Assert the same handler is selected") {
+                        val fixture = findRunDialog().find<ContainerFixture>(byXpath("//div[@class='HandlerPanel']"))
+                        assertThat(fixture.findAllText().joinToString("") { it.text }).isEqualTo("helloworld.App::handleRequest")
+                    }
+                    step("Assert the run configuration has an error (since the project was not imported)") {
+                        assertThat(findRunDialog().findAllText()).anySatisfy { assertThat(it.text).contains("Error") }
                     }
-                    // TODO the field is a JTextComponent so findAllText does not work and we don't have a JTextComponent fixture
-                    // step("Assert the input is the same") {
-                    //    assertThat(dialog.findAllText()).anySatisfy { assertThat(it.text).isEqualTo(input) }
-                    // }
                 }
             }
         }
     }
 
     private fun ContainerFixture.functionModel(): ComboBoxFixture =
         find(byXpath("//div[@class='TextFieldWithBrowseButton']/following-sibling::div[@class='ComboBox']"))
+
+    private fun ContainerFixture.findRunDialog() = find<DialogFixture>(DialogFixture.byTitleContains("Run"), Duration.ofSeconds(5))
```

Comment:
can we be more specific on the title here lol

Reply:
We could but it's more likely to break in the future. Since it's based off the remote IDE, we can't just pull the message from the IDE library, we would have to execute it remotely. Plus, what else would start with "Run" that would likely to be open :P